### PR TITLE
Tag fetcher: allow to apply cover and tags separately // var.1 checkboxes

### LINKED ISSUE #12532: Fetched cover art from "MusicBrainz" is auto applied
### Bug Description

Mixxx 2.4.0-beta
Git Version: 2.4-beta-314-g71dceb27ba
Platform: Linux x86_64
OS: Debian 11

**Preferences -> Library -> "Synchronize library track metadata from/to file tags" is checked (enabled).**

When fetching tags from "MusicBrainz", cover art is also fetched after selecting one of the suggestions.
Clicking on "Apply" auto applies the cover art along with the tags without requiring a click on "Apply Cover".

After manually clearing the auto applied cover art (which I do not want), the downloaded image is still in the filesystem and not removed.

Shouldn't "Apply" be renamed to "Apply Tags" and its functionality changed to only apply tags?
How can I only apply the tags without applying the suggested cover art?

### Version

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by ronso0**
Fixes #12532 

Solution with two checkboxes = two new tr strings
Check states are forgotten when the dialog is closed.
image

With this new tr strings are inevitable. With #12542 not necessarily.

Both solutions probably fit different use cases:
* this: more suitable for batch tagging with cover art if users prefer to stick to the cover art referenced by the MusicBrainz release
   * check the boxes as desired
   * go through tracks, apply
* #12542: better for individual tagging
   * select tags
   * apply tags
   * (optional: select another cover)
   * apply cover
   * = no need to adjust checkboxes if the found covers is not desired

I have no preference since I rarely use this feature, though I think separate buttons are easier to understand (even if we stick to 'Apply') than checkboxes.

### CODE REVIEW

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you. This one matches my use case more. 
Did you consider to make the checkboxes persistent?

File: src/library/dlgtagfetcher.cpp [hunk 1]
```diff
@@ -266,7 +290,22 @@ void DlgTagFetcher::slotTrackChanged(TrackId trackId) {
     }
 }
 
-void DlgTagFetcher::applyTagsAndCover() {
+void DlgTagFetcher::apply() {
+    bool bApplyTags = checkBoxTags->isChecked();
+    bool bApplyCover = checkBoxCover->isChecked();
+    if (bApplyTags) {
+        applyTags();
+        statusMessage->setText(tr("Metadata applied"));
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/library/dlgtagfetcher.cpp:298]: Sure, I'd also prefer "Tags" and can change that, too, than it's consistent at least in this dialog.
(the track menu and the preferences use "Metadata")

And, no, we can not have them "here" literaly, see the comment below.

**Comment by daschuer** [comment]
It work good enough for a first merge. Thank you. 
During test I had the expectation that the cover is cleared when I try to apply a not found cover. Probably not a use case. 
Do we want to ray out the box in this case or remove the cover.

**Comment by ronso0** [comment]
> During test I had the expectation that the cover is cleared when I try to apply a not found cover. Probably not a use case.
Do we want to ray out the box in this case or remove the cover.

I considered that first but I'm not 100% sure the disabled checkbox clearly communicates there's no cover. But that's just my opinion.
I see that Search/Replace dialogs like in Eclipse use the same pattern, e.g. in Eclispe "whole word" is greyed out if "Regex" is checked, so we may implement it that way. Hopefully it will reassure users not to remove the existing cover when there was not cover found for the selected release.

**Comment by daschuer** [comment]
> Hopefully it will reassure users not to remove the existing cover when there was not cover found for the selected release.

Ah yes. So let's do it that way around. If the selected row has no cover, gray out the checkbox + the label. If they want to remove cover they can do it in the track property dialog.

**Comment by ronso0** [comment]
Alright.
> gray out the checkbox + the label.

Which label do you refer to?
The 'Found Cover' image is already the placeholder in that case (and it has no menu).

**Comment by daschuer** [comment]
Oh sorry, I just mean the checkbox label which is grayed out anyway.

Do we need in addition a few lines that also unchecks the greyed out checkbox and restores the previous value after a line with a cover is selected?

**Comment by ronso0** [comment]
Yes, unchecking the inactive box makes sense. I didn't do that previously to not make this quick fix more complex, but not that we agreed on that I'll implement it.

**Comment by ronso0** [comment]
Okay, done. Checkboxes are now disabled when there are no tags / cover to be applied.

'Apply' is **not** disabled when both are unchecked, tracking this to not falsely enable Apply is overkill.
For the same reason I did not implement un/checking the boxes when they're enabled/disabled.

This is good enough now. Please take a look.

**Comment by ronso0** [comment]
OT: we may consier to move the 'preferred size' combobox from the preferences to the TagFetcher.

**Review by daschuer — APPROVED** [review]
LGTM. 
Yes I agree, it works good. 
We can improve that any time later.

**Pull request merged**